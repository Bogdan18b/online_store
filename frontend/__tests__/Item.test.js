import Item from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: "asfasgdfvfrewgf",
  title: 'shoes',
  price: 60000,
  description: 'really cool snickers',
  image: 'shoe.jpg',
  largeImage: 'shoe.jpg',
};

describe('<Item />', () => {
  it('renders and displays properly', () => {
    const wrapper = shallow(<Item item={fakeItem}/>);
    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$600');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  });

  it('renders the image properly', () => {
    const wrapper = shallow(<Item item={fakeItem}/>);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<Item item={fakeItem}/>);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});
