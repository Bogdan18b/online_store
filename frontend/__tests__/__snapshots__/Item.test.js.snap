// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="shoes"
    src="shoe.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "asfasgdfvfrewgf",
          },
        }
      }
    >
      <a>
        shoes
      </a>
    </Link>
  </Title>
  <PriceTag>
    $600
  </PriceTag>
  <p>
    really cool snickers
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "asfasgdfvfrewgf",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="asfasgdfvfrewgf"
    />
    <DeleteItem
      id="asfasgdfvfrewgf"
    />
  </div>
</ItemStyles__Item>
`;
